File: BackendCitasMedicas/src/main/java/com/example/BackendCitasMedicas/Model/Consulta.java
package com.example.BackendCitasMedicas.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Consulta {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id_consulta;
    private LocalDate fecha_consulta;
    private String motivo;
    private String diagnostico;
    private String notas;

    @ManyToOne
    @JoinColumn(name = "id_paciente")
    private Paciente paciente;

    @ManyToOne
    @JoinColumn(name = "id_medico")
    private Medico medico;

    @OneToMany(mappedBy = "consulta")
    @JsonIgnore
    private List<Tratamiento> tratamientos;
}

File: FrontendCitasMedicas/app/src/main/java/com/example/frontcitasmedicas/Model/Consulta.kt
package com.example.frontcitasmedicas.Model

import java.time.LocalDate

data class Consulta(
    val id_consulta: Long = 0,
    val fecha_consulta: String = "", // O `LocalDate` si decidimos implementar conversión
    val motivo: String = "",
    val diagnostico: String = "",
    val notas: String = "",
    val paciente: Paciente? = null,
    val medico: Medico? = null,
    val tratamientos: List<Tratamiento> = emptyList()
)
